packages/next-swc/crates/next-core/src/app_structure.rs

```diff
@@ -697,7 +697,9 @@ async fn directory_tree_to_loader_tree(
 
     components.metadata.base_page = Some(app_page.clone());
 
-    if app_page.is_root() && components.not_found.is_none() {
+    if (app_page.is_root() || (app_path.is_root() && components.layout.is_some()))
```

Comment:
It's hard to understand the order of operations here, seems there's a set of parenthesis missing?

```
(
    app_page.is_root() || 
    (app_path.is_root() && components.layout.is_some()) && 
    components.not_found.is_none()
)
```

The first condition `app_page.is_root() || ` also has to check if `not_found.is_none()` so maybe something like:

```
(
    (
    app_page.is_root() || 
    (app_path.is_root() && components.layout.is_some())
    ) && 
    components.not_found.is_none()
)
```

Even if it already handles that correctly (I'm not sure if it does) the parenthesis will make it clearer what the order is.

Reply:
It's currently unclear to me if `components.not_found.is_none()` is also checked for the `app_page.is_root()` case, just looking at it's really hard to know for certain it is checking that for the first condition as `||` and `&&` are combined

Reply:
Isn't `app_page.is_root()` and `app_path.is_root()` the same?

Reply:
@sokra no, route groups (and parallel routes) are only represented in the `app_page`, `app_path` is the url path which doesn't have route groups in it

Reply:
@timneutkens the `&&` and `||` are separated by parenthesis already, none of the conditions here use operator precedence, but I can try to pull something out